### PR TITLE
Flowtype fix for mincome distribution

### PR DESCRIPTION
**Pull request by ktamas77**

### CODE REVIEW

File: frontend/simple/utils/distribution/mincome-default.js [hunk 1]
```diff
@@ -49,20 +50,27 @@ function transferValue (incomes, indexAmount, transferAmount) {
...
 
 function incomeAmounts (a, b) {
   return a.amount - b.amount
 }
 
-function floorTo (number) {
-  return number.toString().match(/^-?\d+(?:\.\d{0,2})?/)[0]
+function floorTo (number): number {
+  const numberMatch = number.toString().match(/^-?\d+(?:\.\d{0,2})?/)
+  return parseFloat(numberMatch[0])
```
Comment (ktamas77) [inline frontend/simple/utils/distribution/mincome-default.js]: couldn't find the proper flowtype notation for this one - please advise

File: frontend/simple/utils/distribution/mincome-default.js [hunk 2]
```diff
@@ -49,20 +50,27 @@ function transferValue (incomes, indexAmount, transferAmount) {
   })
 }
 
-function incomeKeys (a, b) {
-  return a.name > b.name
+function incomeKeys (a: Object, b: Object): number {
```
Comment (sandrina-p) [inline frontend/simple/utils/distribution/mincome-default.js]: It's a good idea to validate `a` and `b` keys too. Something like:

```
function incomeKeys (a: { name: number }, b: { name: number }: number {
```

File: frontend/simple/utils/distribution/mincome-default.js [hunk 3]
```diff
@@ -49,20 +50,27 @@ function transferValue (incomes, indexAmount, transferAmount) {
...
-function floorTo (number) {
-  return number.toString().match(/^-?\d+(?:\.\d{0,2})?/)[0]
+function floorTo (number): number {
+  const numberMatch = number.toString().match(/^-?\d+(?:\.\d{0,2})?/)
+  return numberMatch ? parseFloat(numberMatch[0]) : 0;
 }
 
-function incomeDistribution (incomes, minCome) {
-  let incomeLength = Object.keys(incomes).length
+function incomeDistribution (incomes: Array<Object>, minCome: number) {
```
Comment (sandrina-p) [inline frontend/simple/utils/distribution/mincome-default.js]: Can we validate the keys inside each object? Something like...

```
incomes: Array<{ amount: number }>
```

Not sure if this is the correct syntax though

File: frontend/simple/utils/distribution/mincome-default.js (hunk 2 above)
Comment (ktamas77) [reply frontend/simple/utils/distribution/mincome-default.js]: nice, thx - although `name: string` is the correct way. i'll add this

File: frontend/simple/utils/distribution/mincome-default.js (hunk 3 above)
Comment (ktamas77) [reply frontend/simple/utils/distribution/mincome-default.js]: yes. done. thanks for the tip!

**Pull request merged**